curator-framework/src/main/java/org/apache/curator/framework/imps/NamespaceImpl.java

```diff
@@ -45,7 +45,7 @@
             }
             catch ( IllegalArgumentException e )
             {
-                throw new IllegalArgumentException("Invalid namespace: " + namespace);
+                throw new IllegalArgumentException("Invalid namespace: " + namespace + ", " + e.getMessage());
```

Comment:
Hmm, it general yes of course. But validatePath did not have such a path. I can correct this if needed?